src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp

```diff
@@ -212,132 +213,39 @@ CorporealityEntry const _corporealityReference[MAX_CORPOREALITY_STATE] = {
-    void EnterCombat(Unit* /*who*/) override
-    {
-        _EnterCombat();
-        me->AddAura(SPELL_TWILIGHT_PRECISION, me);
-        _canEvade = false;
-        events.ScheduleEvent(EVENT_CLEAVE, urand(8000, 10000));
-        events.ScheduleEvent(EVENT_TAIL_LASH, 10000);
-        events.ScheduleEvent(EVENT_BREATH, urand(10000, 15000));
-    }
-
-    void Reset() override
-    {
-        _canEvade = false;
-        _Reset();
-    }
-
-    void JustReachedHome() override
-    {
-        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
-        _JustReachedHome();
-    }
-
-    void ExecuteEvent(uint32 eventId) override
-    {
-        switch (eventId)
-        {
-            case EVENT_CLEAVE:
-                DoCastVictim(SPELL_CLEAVE);
-                events.ScheduleEvent(EVENT_CLEAVE, urand(8000, 10000));
-                break;
-            case EVENT_TAIL_LASH:
-                DoCastAOE(SPELL_TAIL_LASH);
-                events.ScheduleEvent(EVENT_TAIL_LASH, 10000);
-                break;
-            case EVENT_BREATH:
-                DoCast(me, me->GetEntry() == NPC_HALION ? SPELL_FLAME_BREATH : SPELL_DARK_BREATH);
-                events.ScheduleEvent(EVENT_BREATH, urand(10000, 12000));
-                break;
-        }
-    }
-
-    void UpdateAI(uint32 diff) override
-    {
-        if (!UpdateVictim() || me->HasUnitState(UNIT_STATE_CASTING))
-            return;
-
-        events.Update(diff);
-
-        while (uint32 eventId = events.ExecuteEvent())
-            ExecuteEvent(eventId);
-
-        DoMeleeAttackIfReady();
-    }
-
-    void SetData(uint32 index, uint32 dataValue) override
-    {
-        switch (index)
-        {
-            case DATA_EVADE_METHOD:
-                _canEvade = (dataValue == 1);
-                break;
-            default:
-                break;
-        }
-    }
-
-    void SpellHit(Unit* /*who*/, SpellInfo const* spellInfo) override
-    {
-        if (spellInfo->Id == SPELL_TWILIGHT_MENDING)
-            Talk(SAY_REGENERATE);
-    }
-
-protected:
-    bool _canEvade;
-};
-
 class boss_halion : public CreatureScript
 {
     public:
         boss_halion() : CreatureScript("boss_halion") { }
 
-        struct boss_halionAI : public generic_halionAI
+        struct boss_halionAI : public BossAI
         {
-            boss_halionAI(Creature* creature) : generic_halionAI(creature, DATA_HALION)
-            {
-                me->SetHomePosition(HalionSpawnPos);
-            }
-
-            void Reset() override
-            {
-                generic_halionAI::Reset();
-                me->SetReactState(REACT_DEFENSIVE);
-                me->RemoveAurasDueToSpell(SPELL_TWILIGHT_PHASING);
-                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
-            }
+            boss_halionAI(Creature* creature) : BossAI(creature, DATA_HALION) { }
 
             void EnterEvadeMode(EvadeReason why) override
             {
-                if (why == EVADE_REASON_BOUNDARY)
```

Comment:
so far I wasn't able to successfully remove him from the circle, the boundary seems  very sensitive, he always despawn before leaving the circle ^^